docs/pages/userdocs/workflow.dox

```diff
@@ -0,0 +1,83 @@
+/**
+ * \page workflow AqNWB Workflow
+ *
+ * \tableofcontents
+ *
+ * \section recording_workflow Overview of a recording workflow
+ *
+ * For users wanting to integrate NWB with a particular data acquisition software, here
+ * we outline the steps for a single recording from file creation to saving.
+ *
+ * 1. Create the I/O object.
+ * 2. Create the `RecordingContainers` object. 
+ * 3. Create the `NWBFile` object.
+ * 4. Create datasets and add to RecordingContainers.
+ * 5. Start the recording.
+ * 6. Write data.
+ * 7. Stop the recording and close the NWBFile.
```

Comment:
```suggestion
 * For users wanting to integrate NWB with a particular data acquisition software, here
 * we outline the steps for a single recording from file creation to saving.
 *
 * 1. Create the \ref AQNWB::HDF5::HDF5IO "HDF5IO" object used for writing data to the file on disk.
 * 2. Create the \ref AQNWB::NWB::RecordingContainers "RecordingContainers" object used for managing \ref AQNWB::NWB::Container "Container" objects for storing recordings.
 * 3. Create the \ref AQNWB::NWB::NWBFile "NWBFile" object used for managing and creating NWB file contents.
 * 4. Create the \ref AQNWB::NWB::Container "Container" objects (e.g., \ref AQNWB::NWB::ElectricalSeries "ElectricalSeries") used for recording and add them to the \ref AQNWB::NWB::RecordingContainers "RecordingContainers".
 * 5. Start the recording.
 * 6. Write data.
 * 7. Stop the recording and close the \ref AQNWB::NWB::NWBFile "NWBFile".
```

Mainly changing names of classes to references and adding a bit of text to explain what each step/object is used for